homeassistant/components/smhi/config_flow.py

```diff
@@ -12,113 +12,74 @@
+    session = aiohttp_client.async_get_clientsession(hass)
+    smhi_api = Smhi(longitude, latitude, session=session)
+    try:
+        await smhi_api.async_get_forecast()
+    except SmhiForecastException:
+        return False
+
+    return True
 
 
 class SmhiFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     """Config flow for SMHI component."""
 
     VERSION = 1
 
-    def __init__(self) -> None:
-        """Initialize SMHI forecast configuration flow."""
-        self._errors: dict[str, str] = {}
-
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle a flow initialized by the user."""
-        self._errors = {}
 
-        if user_input is not None:
-            is_ok = await self._check_location(
-                user_input[CONF_LONGITUDE], user_input[CONF_LATITUDE]
-            )
-            if is_ok:
-                name = slugify(user_input[CONF_NAME])
-                if not self._name_in_configuration_exists(name):
-                    latitude = user_input[CONF_LATITUDE]
-                    longitude = user_input[CONF_LONGITUDE]
-                    await self.async_set_unique_id(f"{latitude}-{longitude}")
-                    self._abort_if_unique_id_configured()
-                    return self.async_create_entry(
-                        title=user_input[CONF_NAME], data=user_input
-                    )
-
-                self._errors[CONF_NAME] = "name_exists"
-            else:
-                self._errors["base"] = "wrong_location"
-
-        # If hass config has the location set and is a valid coordinate the
-        # default location is set as default values in the form
-        if (
-            not smhi_locations(self.hass)
-            and await self._homeassistant_location_exists()
-        ):
-            return await self._show_config_form(
-                name=HOME_LOCATION_NAME,
-                latitude=self.hass.config.latitude,
-                longitude=self.hass.config.longitude,
-            )
-
-        return await self._show_config_form()
-
-    async def _homeassistant_location_exists(self) -> bool:
-        """Return true if default location is set and is valid."""
-        # Return true if valid location
-        return (
-            self.hass.config.latitude != 0.0
-            and self.hass.config.longitude != 0.0
-            and await self._check_location(
-                self.hass.config.longitude, self.hass.config.latitude
-            )
-        )
+        errors: dict[str, str] = {}
 
-    def _name_in_configuration_exists(self, name: str) -> bool:
-        """Return True if name exists in configuration."""
-        return name in smhi_locations(self.hass)
+        if user_input is not None:
+            lat = user_input[CONF_LATITUDE]
+            lon = user_input[CONF_LONGITUDE]
+            if await async_check_location(self.hass, lon, lat):
+                name = DEFAULT_NAME
```

Comment:
It probably makes sense to put the cords in the name in non-home location case so they know which one to rename
<img width="307" alt="Screen Shot 2022-01-16 at 10 11 22" src="https://user-images.githubusercontent.com/663432/149676308-50a57d16-5ced-445b-846d-62109ea6769e.png">

Reply:
I agree. I also applied some rounding given that the upstream package only deals with 6 decimals anyhow.